libraries/AP_GPS/AP_GPS_UBLOX.cpp

```diff
@@ -1307,19 +1307,26 @@ AP_GPS_UBLOX::broadcast_configuration_failure_reason(void) const {
 /*
   return velocity lag in seconds
  */
-float AP_GPS_UBLOX::get_lag(void) const
+bool AP_GPS_UBLOX::get_lag(float &lag_sec) const
 {
     switch (_hardware_generation) {
+    case UBLOX_UNKNOWN_HARDWARE_GENERATION:
+        lag_sec = 0.22f;
+        // always bail out in this case, it's used to indicate we have yet to recieve a valid
+        // hardware generation
+        return false;
     case UBLOX_5:
     case UBLOX_6:
     default:
-        return 0.22f;
+        lag_sec = 0.22f;
+        break;
     case UBLOX_7:
     case UBLOX_M8:
         // based on flight logs the 7 and 8 series seem to produce about 120ms lag
-        return 0.12f;
+        lag_sec = 0.12f;
         break;
     };
+    return !(_unconfigured_messages & CONFIG_VERSION);
```

Comment:
Why depend on this? If we have the hardware generation, then why not always true? And once again, I'm not sure how this interacts with auto config turned off.

I'm also a bit confused on how we manage the version, we seem to request it by two different ways: MSG_NAV_SVINFO and MSG_MON_VER. But we do different things for both. Specially when receiving MSG_MON_VER we don't remove the CONFIG_VERSION from the unconfigured messages, we wait for it to be requested again to remove it.
My feeling is that we should always request the version instead of being a step of configuration.